### PR TITLE
Reworked InlineTaskApiService using multiprocessing

### PR DESCRIPTION
**Pull request by Wiezzel**
Subprocesses can be killed so InlineTaskApiService supports stop now.

### CODE REVIEW

File: python/golem_task_api/testutils/inlinetaskapiservice.py [hunk 1]
```diff
@@ -22,43 +22,53 @@ def __init__(
...
         self._provider_handler = provider_handler
         self._provider_lifecycle_handler = provider_lifecycle_handler
-        self._thread = None
 
-    def _spawn(self, command: str):
+        self._server_process: Optional[multiprocessing.Process] = None
+
+    def _spawn(self, command: str) -> None:
         loop = asyncio.new_event_loop()
+        loop.add_signal_handler(signal.SIGTERM, loop.stop)
```
Comment (mfranciszkiewicz) [inline python/golem_task_api/testutils/inlinetaskapiservice.py]: I suggest doing the same for `signal.SIGHUP` and `signal.SIGINT`

File: python/golem_task_api/testutils/inlinetaskapiservice.py [hunk 2]
```diff
@@ -22,43 +22,53 @@ def __init__(
...
             command.split(' '),
             requestor_handler=self._requestor_handler,
+            requestor_lifecycle_handler=self._requestor_lifecycle_handler,
             provider_handler=self._provider_handler,
+            provider_lifecycle_handler=self._provider_lifecycle_handler
         ))
 
+    def _join(self) -> None:
+        assert self._server_process is not None
+        self._server_process.join()
```
Comment (mfranciszkiewicz) [inline python/golem_task_api/testutils/inlinetaskapiservice.py]: Perhaps pass a `timeout` value?

File: python/golem_task_api/testutils/inlinetaskapiservice.py (hunk 1 above)
Comment (Wiezzel) [reply python/golem_task_api/testutils/inlinetaskapiservice.py]: We're using `.terminate()` to stop the process which sends `SIGTERM`. Why would the other handlers be needed?

File: python/golem_task_api/testutils/inlinetaskapiservice.py (hunk 2 above)
Comment (Wiezzel) [reply python/golem_task_api/testutils/inlinetaskapiservice.py]: Yea, why not.

File: python/golem_task_api/testutils/inlinetaskapiservice.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply python/golem_task_api/testutils/inlinetaskapiservice.py]: It's not like we need them, they can be simply sent to the process. Not insisting.

File: python/tests/test_integration.py [hunk 3]
```diff
@@ -20,44 +20,43 @@ class AsyncMock(mock.MagicMock):
...
     resources = []
     task_params = {'test_param': 'test_value'}
 
+    # Mocks have to be configured before calling init_requestor_with_handler
+    # because it creates copies of these objects when spawning server process.
+    requestor_handler.create_task.return_value = task_struct
+    requestor_handler.has_pending_subtasks.return_value = True
+    requestor_handler.next_subtask.return_value = \
+        structs.Subtask('subtask_id', {}, [])
+    result_path = Path('result')
```
Comment (mfranciszkiewicz) [inline python/tests/test_integration.py]: Kinda lost in-between all those mocks :)

Comment (Wiezzel) [reply python/tests/test_integration.py]: Not anymore.

**Pull request merged**